src/browser/cached_resource_fetcher.ts

```diff
@@ -226,14 +229,22 @@ function download(fileUrl: string, filePath: string): Promise<any> {
         });
 
         request.on('login', (authInfo: any, callback: AuthCallback) => {
-            authRequest(filePath, authInfo, callback);
+            authRequest(saveToPath, authInfo, callback);
         });
 
         request.once('error', (err: Error) => {
             reject(err);
         });
 
-        request.end();
+        if (session) {
+            session.cookies.get({}, (error, cookies) => {
+                const cookiesNameValue = cookies.map((e) => `${e.name}=${e.value}`);
```

Comment:
What if there is an error? Should we be handling, rejecting, or we don't care?

Reply:
Thanks for pointing this out, missed it, will add a check